sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/datasources/subscriptions/SubscriptionsDbConnector.groovy

```diff
@@ -11,15 +20,99 @@ import life.qbic.business.project.subscribe.SubscriptionDataSource
  * @since <version tag>
  */
 class SubscriptionsDbConnector implements SubscriptionDataSource {
+  
+  private final ConnectionProvider connectionProvider
+  
+      /**
+       * Creates a database connector that will connect to the database using the provided connection
+       * provider.
+       * @param connectionProvider the connection provider providing the connections that are used to
+       * connect to the database
+       * @since 1.0.0
+       */
+      SubscriptionsDbConnector(ConnectionProvider connectionProvider) {
+          this.connectionProvider = connectionProvider
+      }
+
     /**
-     * Creates the subscriber in the data source and subscribes him to the project.
+     * Creates the subscriber in the data source, if they don't exist, and subscribes them to the project.
      * @param subscriber
      * @param projectCode
      * @since 1.1.0
      */
     @Override
     void subscribeToProject(Subscriber subscriber, String projectCode) {
-        //TODO implement
-        println "subscriber = $subscriber, projectCode = $projectCode"
+          try {
+            Connection connection = connectionProvider.connect()
+            connection.setAutoCommit(false)
+      
+            connection.withCloseable { it ->
+              try {
+                int subscriberId = addSubscriber(it, subscriber)
+                addSubscription(it, subscriberId, projectCode)
+                connection.commit()
+              } catch (Exception e) {
+                log.error(e.message)
+                log.error(e.stackTrace.join("\n"))
+                connection.rollback()
+                throw new DataSourceException("Subscription to {$projectCode} could not be created: {$subscriber}")
+              } finally {
+                connection.close()
+              }
+            }
+          } catch (Exception e) {
+            log.error(e)
+            log.error(e.stackTrace.join("\n"))
+            throw new DataSourceException("Subscription to {$projectCode} could not be created: {$subscriber}")
+          }
+    }
+    
+    private int addSubscriber(Connection connection, Subscriber subscriber) {
```

Comment:
Is it addSubscriber or getSubscriberId?

Reply:
I mean, you want to get the subscriber ID in order to subscribe a subscriber, and it's a "corner case" that some of the people need to be added as subscribers(major NIT). But changing the method naming could help to improve code readability